Makefile

```diff
@@ -108,10 +108,19 @@ verify-kustomize-repo: \
 prow-presubmit-check: \
 	install-tools \
 	test-unit-kustomize-plugins \
-	test-go-mod \
 	build-non-plugin-all \
 	test-examples-kustomize-against-HEAD \
 	test-examples-kustomize-against-v4-release
+	# TODO(annasong): Find permanent solution.
+	# We have temporarily commented out this job because it fails for public
+	# cross-module changes, given that we've removed replace directives with go
+	# workspace and the job runs on the latest released modules. This
+	# complicates releases.
+	# This job is also less effective than intended because it only detects
+	# go mod tidy errors, instead of any changes to the original code that
+	# go mod tidy makes.
+
+	# test-go-mod
```

Comment:
I've been thinking about this, and I wonder if it is going to cause us other problems to solve it this way. Namely, say we need to update a dependency in the problematic module in the interim--if I understand correctly, we won't be able to run `go mod tidy` on that module without manually adding and then removing replace statements... 

🤔 Might it work to have the make target do that for us?  i.e. change the part that runs in each module to be a short script that uses `go mod edit -replace` for each of our released modules, then `go mod tidy`, then `go mod edit -dropreplace` for each released module? `hack/for-each-module` exposes `KUSTOMIZE_ROOT`, which should make it easier to find the right path for those commands.

Reply:
Btw I opened https://github.com/kubernetes-sigs/kustomize/issues/4977 for the problem of the CI task not really fulfilling its purpose currently